sql/parser/normalize.go

```diff
@@ -119,19 +122,25 @@ func (expr *ComparisonExpr) normalize(v *normalizeVisitor) Expr {
 				//   /     \            /     \
 				//  a       1          2       1
 
+				rotating := true
 				switch left.Operator {
-				case Plus, Minus, Div:
+				case Plus:
+					left.Operator = Minus
+				case Minus:
+					left.Operator = Plus
+				case Div:
+					left.Operator = Mult
+				default:
+					rotating = false
+				}
+
+				if rotating {
+					// Clear the function caches since we're rotating.
+					expr.fn.fn = nil
+					left.fn.fn = nil
+
```

Comment:
Ditto about equivalence. I don't find this any more (or less) readable. Wondering why bother.